extensions/wikia/AdEngine/js/MonetizationServiceHelper.js

```diff
@@ -0,0 +1,144 @@
+define('ext.wikia.adEngine.monetizationsServiceHelper', [
+	'jquery',
+	'wikia.geo',
+	'wikia.scriptwriter',
+	'wikia.tracker',
+], function ($, geo, scriptWriter, tracker) {
+	'use strict';
+
+	var track = tracker.buildTrackingFunction({
+			trackingMethod: 'internal',
+			category: 'monetization-module',
+			geo: geo.getCountryCode()
+		});
+
+	/**
+	 * @desc Inject content to the slot
+	 * @param {string} slot
+	 * @param {string} content
+	 * @param {function} success
+	 */
+	function injectContent(slot, content, success) {
+		scriptWriter.injectHtml(slot, content, function () {
+			var $container = $('#' + slot).find('.monetization-module');
+
+			trackImpression($container);
+			trackClickEvent($container);
+
+			success();
+		});
+	}
+
+	/**
+	 * @desc Get maximum number of ads for the page
+	 * @returns {number}
+	 */
+	function getMaxAds() {
+		var pageHeight = $('#WikiaArticle').height();
+		if (pageHeight > 5000) {
+			return 3;
+		} else if (pageHeight > 1500) {
+			return 2;
+		}
+		return 1;
+	}
+
+	/**
+	 * @desc Get country code
+	 * @returns {string}
+	 */
+	function getCountryCode() {
+		var countryCodeROW = 'ROW',
+			countryCodes = ['AU', 'CA', 'DE', 'GB', 'HK', 'JP', 'MX', 'RU', 'TW', 'US'],
+			countryCode = geo.getCountryCode();
+		if ($.inArray(countryCode, countryCodes) >= 0) {
+			return countryCode;
+		} else {
+			return countryCodeROW;
+		}
+	}
+
+	/**
+	 * @desc Track impression
+	 */
+	function trackImpression($container) {
+		// track impression for each placement
+		$container.each(function () {
+			var $this = $(this),
+				value = $this.children().children().length,	// check if the ad is blocked
```

Comment:
When ad blocker is enabled, the ad is not rendered even though there is an ad for the slot.